src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/OrderedHashtable.cs

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Specialized;
+using System.Runtime.Serialization;
+
+#nullable enable
+
+namespace Microsoft.PowerShell.Commands
+{
+    /// <summary>
+    /// OrderedHashtable is a hashtable that preserves the order of the keys.
+    /// </summary>
+    public class OrderedHashtable : Hashtable, IEnumerable
+    {
+        private readonly OrderedDictionary _orderedDictionary;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OrderedHashtable"/> class.
+        /// </summary>
+        public OrderedHashtable()
+        {
+            _orderedDictionary = new OrderedDictionary();
```

Comment:
Although the intention is clear this substitutes one API for another, which can lead to unpleasant amazement of users.